src/SSVOpenHexagon/Core/HexagonGame.cpp

```diff
@@ -122,20 +113,27 @@ HexagonGame::HexagonGame(Steam::steam_manager& mSteamManager,
         initKeyIcons();
     };
 
-    add2StateInput(game, Config::getTriggerRotateCW(), inputImplCW);
-    add2StateInput(game, Config::getTriggerRotateCCW(), inputImplCCW);
-    add2StateInput(game, Config::getTriggerFocus(), inputFocused);
-    add2StateInput(game, Config::getTriggerSwap(), inputSwap);
+    // keyboard binds
+    Config::keyboardBindsSanityCheck();
+    
+    add2StateInput(game, Config::getTriggerRotateCCW(), inputImplCCW, Tid::RotateCCW);
+    add2StateInput(game, Config::getTriggerRotateCW(), inputImplCW, Tid::RotateCW);
+    add2StateInput(game, Config::getTriggerFocus(), inputFocused, Tid::Focus);
+    add2StateInput(game, Config::getTriggerSwap(), inputSwap, Tid::Swap);
+    game.addInput(
+        {{sf::Keyboard::Key::Escape}}, [this](ssvu::FT /*unused*/) { goToMenu(); }, //hardcoded
+        ssvs::Input::Type::Always);
     game.addInput(
-        Config::getTriggerExit(), [this](ssvu::FT /*unused*/) { goToMenu(); });
+        Config::getTriggerExit(), [this](ssvu::FT /*unused*/) { goToMenu(); },              //editable
+        ssvs::Input::Type::Always, Tid::Exit);
     game.addInput(
         Config::getTriggerForceRestart(),
         [this](ssvu::FT /*unused*/) {
             status.mustStateChange = StateChange::MustRestart;
         },
-        ssvs::Input::Type::Once);
+        ssvs::Input::Type::Once, Tid::ForceRestart);
     game.addInput(
-        Config::getTriggerRestart(),
+        {{sf::Keyboard::Key::R}},                                                           //hardcoded
```

Comment:
Why do we want to hardcode `R`?

Reply:
I made it hardcoded because it is the key that player is told to be pressed to restart after it dies